### PR TITLE
Add Module and Availability to ModelClassDiagram 

### LINKED ISSUE #60: Update Guides for Module.java :file_folder:
- [x] DeveloperGuide for Module.java
- ~~UserGuide for Module.java~~

### LINKED ISSUE #63: Update Guides for Availability.java :file_folder:
- [x] DeveloperGuide for Availability.java
- ~~UserGuide for Availability.java~~

### PR DESCRIPTION
**Pull request by whitesnowx**
**(Not expected to, but if this creates a merge conflict with the previous pr, ignore this pr, will close)**

Update Developer Guide diagram -> ModelClassDiagram

### CODE REVIEW

**Comment by whitesnowx** [comment]
Not sure where else needs to update the developer guide for Module and Availability, support requestedd

File: docs/diagrams/ModelClassDiagram.puml
Comment (iynixil) [inline docs/diagrams/ModelClassDiagram.puml:1]: Looks good to me! Although this is to update "module" and "availability", looks like the other attributes were added.
Maybe "Faculty" can be added as well?

Comment (whitesnowx) [reply docs/diagrams/ModelClassDiagram.puml:1]: The other attributes were copied during the initial fork.

@JerryWang0000 would you prefer to include faculty together here or would you prefer to do it on your own?

Comment (JerryWang0000) [reply docs/diagrams/ModelClassDiagram.puml:1]: I do it on my own

**Review by JerryWang0000 — APPROVED** [review]

**Pull request merged**